common/util/file_util.cc

```diff
@@ -174,7 +174,18 @@ absl::Status SetContents(absl::string_view filename,
 }
 
 std::string JoinPath(absl::string_view base, absl::string_view name) {
-  return absl::StrCat(base, "/", name);
+  // TODO: this will certainly be different on Windows
+  static constexpr absl::string_view kFileSeparator = "/";
+
+  // Make sure that we don't concatenation multiple superfluous separators.
+  // Note, since we're using string_view, the substr() operations are cheap.
+  while (!base.empty() && base[base.length() - 1] == kFileSeparator[0]) {
+    base = base.substr(0, base.length() - 1);
+  }
```

Comment:
It is not immediately clear that this will work if base is just `"/"` - but I guess it will as the separator will be the leading `/` then - maybe we could add a comment saying why it's okay?

Reply:
I solved it by adding a test to make sure that this corner-case is covered, since anyone wondering about the implementation will look there first.